powersimdata/input/grid.py

```diff
@@ -29,44 +29,20 @@ class Grid:
     :raises ValueError: if source or engine does not exist.
     """
 
-    def __init__(self, interconnect, source="usa_tamu", engine="REISE"):
+    def __init__(self, model_data):
         """Constructor."""
-        if not isinstance(source, str):
-            raise TypeError("source must be a str")
-        if source not in self.SUPPORTED_MODELS and not source.endswith(".mat"):
-            raise ValueError(
-                f"Source must be one of {','.join(self.SUPPORTED_MODELS)} "
-                "or the path to a .mat file that represents a grid "
-            )
-        if engine not in self.SUPPORTED_ENGINES:
-            raise ValueError(
-                f"Engine must be one of {','.join(self.SUPPORTED_ENGINES)}"
-            )
-
-        key = cache_key(interconnect, source)
-        cached = _cache.get(key)
-        if cached is not None:
-            data = cached
-        elif source == "usa_tamu":
-            data = TAMU(interconnect)
-        elif os.path.splitext(source)[1] == ".mat":
-            if engine == "REISE":
-                data = FromREISE(source)
-            elif engine == "REISE.jl":
-                data = FromREISEjl(source)
```

Comment:
Ultimately, I'd like to move the initialization of the `data` variable _outside_ of this `Grid` class. This will make the class more maintainable on the long run. If you add 30 more models, you'll have 30 branching if statements here, vs if you initialize the `data` variable outside of the class and pass it in, you would just have
`Grid(TAMU(interconnect))` or `Grid(OTHERMODEL(interconnect))`.

Reply:
I agree. What do you think @jon-hagg, @danielolsen, @BainanXia and @ahurli? I think that it could also solve the issue where a grid model (e.g., a single small country) has no interconnection.

Reply:
+1

Reply:
I think every grid model should have at least one interconnection.

Reply:
An interconnection defines a power grid that is connected by ac branches synchronized with same frequency when solving OPF. If every bus is connected by a path with only ac lines with every other bus in the network, it is one interconnection, no matter it is big or small.

Reply:
So an interconnection can be a country name then?

Reply:
There's no reason why not.

Reply:
I thought it would be weir to have, e.g., a `Australia` class that we would instantiate through `Australia("Australia")` (I am sure Australia has several interconnections and it is a bad example). In that case I would rather do `Australia()`

Reply:
Should it be model name instead, something like `Good_Australia_Model("Australia")`, just like `TAMU("USA")`?

Reply:
> I think every grid model should have at least one interconnection.
@danielolsen just to be clear, this is in response to @rouille's statement "I think that it could also solve the issue where a grid model (e.g., a single small country) has no interconnection.", or was it for/against the proposal I made?

Reply:
Also, just to verify my understanding, it's `TAMU(interconnect)`, `FromREISE(source)`, and `FromREISEjl(source)` that are the very expensive time consuming function calls right?

Reply:
@goccert25 yes, I was referring to @rouille's comment.

`TAMU(interconnect)` involves reading and parsing a few hundred thousand lines of CSVs, and `FromREISE(source)` (almost never used anymore) and `FromREISEjl(source)` involve reading and parsing a matfile, and for these I think the lengthiest part of the operation is the `powersimdata.input.scenario_grid.format_gencost` function.

Reply:
@rouille from the user side, we would still have `Grid(interconnect="Australia", source="Good_Australia_Model")`, right? If we define the `Good_Australia_Model` class appropriately, we should be able to get the hypothetical single interconnect via `Good_Australia_Model()` I think. I guess we could make `interconnect` an optional parameter in some cases, and then it would be possible to call `Grid(source="Good_Australia_Model")`.

Reply:
Yeah. My comment was related to the new instantiation that is discussed here: `Grid(Australia("Australia"))`. If model = country name = interconnection, it seems better to do `Grid(Australia())` or even `Grid(Australia())` if there is only one interconnection and the grid model is explicit.

Reply:
Ah, I see now, I got a little lost in the conversation.

I guess if we assume that the only grid model that can be instantiated with just a interconnect name (or a list of them) and nothing else is the `usa_tamu` model, then we can take a single parameter for a `Grid` init: which can be either an `AbstractGrid` instance (i.e. `TAMU("Eastern")` or `OTHERMODEL(interconnect)` or `FromREISEjl(filepath)`) or a list/str, which will get passed to a `TAMU` object, so `Grid("Eastern")` will become `Grid(TAMU("Eastern"))`, etc.